src/engine/GameWorker.ts

```diff
@@ -139,10 +141,18 @@ async function onInit({
     return interval;
   };
 
-  gameLoop();
+  if (isFirefox) {
+    update(ctx);
```

Comment:
Rather than trying to reuse the `update` method and doing the useragent check every frame, I'd create two loop functions. One that uses `setInterval` and one that uses `setTimeout` both would call a common `update` function that excludes scheduling the `setTimeout` and just updates the timers and runs the systems.